rest/bootstrap_test.go

```diff
@@ -165,6 +165,23 @@ func adminRequest(t *testing.T, method, path, body string) *http.Response {
 	return resp
 }
 
+func adminRequestWithHeaders(t *testing.T, method, path, body string, headers map[string]string) *http.Response {
```

Comment:
I named `adminRequest` above quite poorly, it's a bit too easy to misuse outside of the context of a bootstrap test. Would suggest this to avoid accidental usage (the 4985+bootstrapTestPortOffset) thing is very much specific to one small set of tests.

```suggestion
func bootstrapAdminRequestWithHeaders(t *testing.T, method, path, body string, headers map[string]string) *http.Response {
```

Reply:
Also opted to rename the existing adminRequest. I know it doesn't make the PR quite as clean to review but as you say, the naming can be improved so decided to do that here.